libcloudforensics/providers/kubernetes/mitigation.py

```diff
@@ -30,40 +32,49 @@ def DrainWorkloadNodesFromOtherPods(
         to prevent pods from appearing on the nodes again as it will be marked
         as unschedulable. Defaults to True.
   """
-  nodes = set(pod.GetNode() for pod in workload.GetCoveredPods())
+  nodes = workload.GetCoveredNodes()
   if cordon:
     for node in nodes:
       node.Cordon()
   for node in nodes:
-    node.Drain(lambda p: not workload.IsCoveringPod(p))
+    node.Drain(lambda pod: not workload.IsCoveringPod(pod))
 
 
-def CreateDenyAllNetworkPolicyForWorkload(
+def IsolatePodsWithNetworkPolicy(
     cluster: k8s.K8sCluster,
-    workload: workloads.K8sWorkload) -> netpol.K8sDenyAllNetworkPolicy:
-  """Isolates a workload's pods via a deny all network policy.
+    pods: List[base.K8sPod]) -> Optional[netpol.K8sDenyAllNetworkPolicy]:
+  """Isolates pods via a deny-all NetworkPolicy.
+
+  **Warning:** It is the caller's responsibility to make sure that Kubernetes
```

Comment:
This should probably be logged as a warning in the method, because I don't expect people to read this (esp. those that are using the lib through `pip install`